tools/mpy_ld.py

```diff
@@ -1207,13 +1215,26 @@ def link_objects(env, native_qstr_vals_len):
             sym.section = env.obj_table_section
         elif sym.name in env.known_syms:
             sym.resolved = env.known_syms[sym.name]
+        elif sym.name in env.externs:
+            # Fixed-address symbols do not need pre-processing.
+            continue
         else:
             if sym.name in fun_table:
                 sym.section = mp_fun_table_sec
                 sym.mp_fun_table_offset = fun_table[sym.name]
             else:
                 undef_errors.append("{}: undefined symbol: {}".format(sym.filename, sym.name))
 
+    if env.externs:
+        for sym in env.unresolved_syms:
+            if sym.name in env.externs:
+                log(
```

Comment:
I'm not sure the logic of this logging is correct.  Eg building `framebuf` I get:
```
$ make ARCH=xtensa -B
GEN build/framebuf_xtensa.config.h
CC framebuf.c
LINK build/framebuf.o
Symbol __udivsi3 is a fixed-address symbol at 4000e21c and is also provided from an object file
Symbol __divsi3 is a fixed-address symbol at 4000dc88 and is also provided from an object file
arch:         EM_XTENSA
text size:    8108
bss size:     160
GOT entries:  29
GEN framebuf_xtensa.mpy
```
This example doesn't use libc, so I don't think these symbols are provided from an object file.

Reply:
That's interesting - I tested that on the `deflate` module and it yielded the correct result as it's one where there was at least a redundant ELF symbol...  But yes, thanks for spotting this.

It is now sorted out, tested on `btree`, `deflate`, and `framebuf` modules.